pyrorl/pyrorl/envs/map_helpers/create_map_info.py

```diff
@@ -0,0 +1,210 @@
+    with open(populated_areas_filename, 'rb') as f:
+        populated_areas = pkl.load(f)
+    paths_filename = os.path.join(map_directory_path, "paths_array.pkl")
+    with open(paths_filename, 'rb') as f:
+        paths = pkl.load(f)
+    paths_to_pops_filename = os.path.join(map_directory_path, "paths_to_pops_array.pkl")
+    with open(paths_to_pops_filename, 'rb') as f:
+        paths_to_pops = pkl.load(f)
+    map_size_and_percent_popualted_list_filename = os.path.join(map_directory_path, "map_size_and_percent_popualted_list.pkl")
+    with open(map_size_and_percent_popualted_list_filename, 'rb') as f:
+        map_size_and_percent_popualted_list = pkl.load(f)
+    num_rows = map_size_and_percent_popualted_list[0]
+    num_cols = map_size_and_percent_popualted_list[1]
+    percent_map_populated = map_size_and_percent_popualted_list[2]
+    return num_rows, num_cols, populated_areas, paths, paths_to_pops, percent_map_populated
+
+
+def generate_map_info(num_rows, num_cols, percent_map_populated, save_map = True):
+    if percent_map_populated > 100:
+        raise Exception("Cannot have more than 100 percent of the map be populated!")
+
+    orientations = create_orientations()
+    directions = {0: "left", 1: "right", 2: "straight"}
+
+    paths_to_pops = {}
+    num_populated_areas = int(num_rows * num_cols * percent_map_populated * 0.01)
+    populated_areas = generate_pop_locations(
+        num_rows, num_cols, num_populated_areas
+    )
+
+    # the number of paths for each populated area is chosen from normal distribution
+    num_paths_array = np.random.normal(3, 1, num_populated_areas).astype(int)
+    # each populated area must have at least one path
+    while 0 in num_paths_array:
+        num_paths_array = np.random.normal(3, 1, num_populated_areas).astype(int)
+
+    paths = []
+    path_num = 0
+
+    for i in range(len(populated_areas)):
+        pop_row, pop_col = populated_areas[i]
+
+        num_pop_paths_created = 0  # for cases where a path couldn't be made
+        while num_pop_paths_created < num_paths_array[i]:
+            current_path = []
+
+            cur_row, cur_col = pop_row, pop_col
+
+            # initialize bounds to not restrict to start
+            x_min = num_rows
+            x_max = -1
+            y_min = num_cols
+            y_max = -1
+
+            # which direction to span out from first
+            orientation = random.choice(["north", "south", "east", "west"])
+
+            done = False
+            while not done:
+
+                # we want to make sure that the current path will not intersect with itself
+                direction_chosen = False
+                num_steps = random.randint(2, 4)
+
+                while not direction_chosen:
+                    # have a bias toward going straight
+                    direction_index = random.randint(0, 1)
+                    if direction_index == 2:
```

Comment:
Based on above `random.randint(0, 1)`, do we know if `direction_index` ever returns a 2?

Reply:
Good catch - this was a change to 0 indexing that I made